File: src/main/scala/com/skafka/example/SimpleProducer.scala
package com.skafka.example
import java.util.Properties

import org.apache.kafka.clients.producer.{ProducerRecord}
import org.apache.kafka.common.errors.WakeupException
import org.apache.kafka.common.serialization.StringSerializer
import org.slf4j.LoggerFactory

class SimpleProducer extends KafkaProducer {
  val logger = LoggerFactory.getLogger(this.getClass)

  private val props = new Properties()
  props.put("bootstrap.servers", "localhost:9092")
  props.put("acks", "all")
  props.put("retries", "5")
  props.put("key.serializer", classOf[StringSerializer].getName)
  props.put("value.serializer", classOf[StringSerializer].getName)

  private val producer = new org.apache.kafka.clients.producer.KafkaProducer[String, String](props)

  def send(topic: String, message: String): Option[Int] = {
    val record = new ProducerRecord[String, String](topic, message)
    try {
      producer.send(record).get()
      logger.info(record.value())
      Some(1)
    }
    catch {
      case wue: WakeupException =>
        logger.debug("Error in reading from kafka ", wue)
        None
      case ex: Exception =>
        logger.error("Error in reading from kafka ", ex)
        None
    }
  }
}
trait KafkaProducer {
  def send(topic : String , message : String) : Any
}

File: src/main/scala/com/kafka/examples/SimpleProducer.java
package com.kafka.examples;

import java.lang.*;

import java.util.Properties;

//import simple producer packages
import org.apache.kafka.clients.producer.Producer;

//import KafkaProducer packages
import org.apache.kafka.clients.producer.KafkaProducer;

//import ProducerRecord packages
import org.apache.kafka.clients.producer.ProducerRecord;

//Create java class named “SimpleProducer”
public class SimpleProducer {

    public static void main(String[] args){
        System.out.println("started testing..........");
        // Check arguments length value
        if(args.length == 0){
            System.out.println("Enter topic name");
            return;
        }

        //Assign topicName to string variable
        String topicName = args[0];

        // create instance for properties to access producer configs
        Properties props = new Properties();

        //Assign localhost id
        props.put("bootstrap.servers", "localhost:9092");
        props.put("acks", "all");
        props.put("retries", 0);
        props.put("batch.size", 16384);
        props.put("linger.ms", 1);
        props.put("buffer.memory", 33554432);
        props.put("key.serializer", "org.apache.kafka.common.serialization.StringSerializer");
        props.put("value.serializer", "org.apache.kafka.common.serialization.StringSerializer");

        Producer<String, String> producer = new KafkaProducer<>(props);

        for(int i = 0; i < 10; i++)
            producer.send(new ProducerRecord<>(topicName, Integer.toString(i), Integer.toString(i)));
        System.out.println("Message sent successfully");
        producer.close();
    }
}